Source/WebCore/platform/SharedBuffer.cpp

```diff
@@ -247,6 +247,10 @@ String FragmentedSharedBuffer::toHexString() const
 
 RefPtr<ArrayBuffer> FragmentedSharedBuffer::tryCreateArrayBuffer() const
 {
+    if (size() > std::numeric_limits<unsigned>::max()) {
+        WTFLogAlways("SharedBuffer::tryCreateArrayBuffer Unable to create buffer. Requested size is too large (%zu)\n", size());
+        return nullptr;
+    }
     auto arrayBuffer = ArrayBuffer::tryCreateUninitialized(static_cast<unsigned>(size()), 1);
```

Comment:
Any idea what the intent of the original author here was in adding the static_cast? ArrayBuffer::tryCreateUninitialized() takes a size_t for the first argument, so it is a bit odd.